### PR TITLE
Clean up `ModalCard::class`.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: docs/modalcard.md [hunk 1]
```diff
@@ -36,54 +42,84 @@ $assetManager->registerMany([
...
+`headerAttributes(array $value)` | HTML attributes for the header. | []
+`id(string $value)` | Widget ID. | `''`
+`modalCardBackgroundClass(string $value)` | The class for the modal background. | `modal-background`
+`modalCardBodyClass(string $value)` | The class for the modal body. | `modal-card-body`
+`modalCardButtonClass(string $value)` | The class for the modal button. | `button modal-button`
+`modalCardClass(string $value)` | The class for the modal card. | `modal-card`
+`modalCardContentClass(string $value)` | The class for the modal card content. | `modal-card-content`
+`modalCardFootClass(string $value)` | The class for the modal card foot. | `modal-card-foot`
+`modalCardHeadClass(string $value)` | The class for the modal card head. | `modal-card-head`
+`modalCardTitleClass(string $value)` | The class for the modal card title. | `modal-card-title`
```
Comment (samdark) [inline docs/modalcard.md:115]: Do we need to add "modalCard" to all these methods?

Comment (terabytesoftw) [reply docs/modalcard.md:115]: There's no need.

Comment (terabytesoftw) [reply docs/modalcard.md]: fixed.

File: docs/modalcard.md [hunk 2]
```diff
@@ -36,54 +42,84 @@ $assetManager->registerMany([
...
-`withoutCloseButton()` | Disable close button. | `false`
-`withoutToggleButton()` | Disable toggle button. | `false`
+`backgroundClass(string $value)` | The class for the modal background. | `modal-background`
+`bodyAttributes(array $value)` | HTML attributes for the body tag.| []
+`bodyClass(string $value)` | The class for the modal body. | `modal-card-body`
+`buttonClass(string $value)` | The class for the modal button. | `button modal-button`
+`cardAttributes(array $value)` | HTML attributes for the card. | []
+`closeButtonAttributes(array $value)` | HTML attributes for the close button. | []
+`closeButtonClass(string $value)` | The class for the close button. | `button delete`
+`closeButtonSize(string $value)` | Size close button. Options available are: (`ModalCard::SIZE_SMALL`, `ModalCard::SIZE_MEDIUM`, `ModalCard::SIZE_LARGE`). | Default setting empty normal.
```
Comment (samdark) [inline docs/modalcard.md:106]:
```suggestion
`closeButtonSize(string $value)` | Сlose button size. Options available are: (`ModalCard::SIZE_SMALL`, `ModalCard::SIZE_MEDIUM`, `ModalCard::SIZE_LARGE`). | Default setting is "normal".
```

File: docs/modalcard.md [hunk 3]
```diff
@@ -36,54 +42,84 @@ $assetManager->registerMany([
...
+`contentClass(string $value)` | The class for the modal card content. | `modal-card-content`
+`footClass(string $value)` | The class for the modal card foot. | `modal-card-foot`
+`footer(string $value)` | The footer content. | `''`
```
Comment (samdark) [inline docs/modalcard.md:109]: Are foot and footer the same thing?

File: docs/modalcard.md [hunk 4]
```diff
@@ -36,54 +42,84 @@ $assetManager->registerMany([
...
+`footerAttributes(array $value)` | HTML attributes for the footer. | []
+`headClass(string $value)` | The class for the modal card head. | `modal-card-head`
+`headerAttributes(array $value)` | HTML attributes for the header. | []
```
Comment (samdark) [inline docs/modalcard.md:112]: Are head and header the same thing?

File: docs/modalcard.md [hunk 5]
```diff
@@ -36,54 +42,84 @@ $assetManager->registerMany([
...
+`id(string $value)` | Widget ID. | `''`
+`modalCardClass(string $value)` | The class for the modal card. | `modal-card`
```
Comment (samdark) [inline docs/modalcard.md:114]: Can we remove "modalCard" prefix?

File: docs/modalcard.md [hunk 6]
```diff
@@ -36,54 +42,84 @@ $assetManager->registerMany([
...
+`title(string $value)` | The title content. | `''`
+`titleAttributes(array $value)` | HTML attributes for the title. | []
+`titleClass(string $value)` | The class for the modal card title. | `modal-card-title`
+`toggleButtonAttributes(array $value)` | HTML attributes for the toggle button. | []
+`toggleButtonColor(string $value)` | Color toggle button. Options available are: (`ModalCard::COLOR_PRIMARY`, `ModalCard::COLOR_INFO`, `ModalCard::COLOR_SUCCESS`, `ModalCard::COLOR_WARNING`, `ModalCard::COLOR_DANGER`, `ModalCard::COLOR_DARK`). | `''`
```
Comment (samdark) [inline docs/modalcard.md:119]:
```suggestion
`toggleButtonColor(string $value)` | Toggle button color. Options available are: (`ModalCard::COLOR_PRIMARY`, `ModalCard::COLOR_INFO`, `ModalCard::COLOR_SUCCESS`, `ModalCard::COLOR_WARNING`, `ModalCard::COLOR_DANGER`, `ModalCard::COLOR_DARK`). | `''`
```

File: docs/modalcard.md [hunk 7]
```diff
@@ -36,54 +42,84 @@ $assetManager->registerMany([
...
+`toggleButtonId(?string $value)` | Toggle button ID. | `''`
+`toggleButtonLabel(string $value)` | Toggle button label. | `'Toggle button'`
+`toggleButtonSize(string $value)` | Size toggle button. Options available are: (`ModalCard::SIZE_SMALL`, `ModalCard::SIZE_MEDIUM`, `ModalCard::SIZE_LARGE`). | Default setting empty normal.
```
Comment (samdark) [inline docs/modalcard.md:122]:
```suggestion
`toggleButtonSize(string $value)` | Toggle button size. Options available are: (`ModalCard::SIZE_SMALL`, `ModalCard::SIZE_MEDIUM`, `ModalCard::SIZE_LARGE`). | Default setting is "normal".
```

File: docs/modalcard.md [hunk 8]
```diff
@@ -36,54 +42,84 @@ $assetManager->registerMany([
...
+`withoutCloseButton(bool $value)` | Whether the close button is disable. | `false`
```
Comment (samdark) [inline docs/modalcard.md:123]:
```suggestion
`withoutCloseButton(bool $value)` | Whether the close button is disabled. | `false`
```

File: docs/modalcard.md [hunk 9]
```diff
@@ -36,54 +42,84 @@ $assetManager->registerMany([
...
+`withoutToggleButton(bool $value)` | Whether the toggle button is disable. | `false`
```
Comment (samdark) [inline docs/modalcard.md:124]:
```suggestion
`withoutToggleButton(bool $value)` | Whether the toggle button is disabled. | `false`
```

File: src/ModalCard.php [hunk 10]
```diff
@@ -42,302 +46,495 @@ final class ModalCard extends Widget
...
 
         return $new;
     }
 
     /**
-     * Returns a new instance with the specified toggle button options.
+     * Returns a new instance with the specified close button size.
      *
-     * @param array $value The toggle button options.
+     * @param string $value The close button size. Default setting empty normal.
```
Comment (samdark) [inline src/ModalCard.php:174]:
```suggestion
     * @param string $value The close button size. Default setting is "normal".
```

File: src/ModalCard.php [hunk 11]
```diff
@@ -42,302 +46,495 @@ final class ModalCard extends Widget
...
+     * Possible values are: ModalCard::SIZE_SMALL, ModalCard::SIZE_MEDIUM, ModalCard::SIZE_LARGE.
+     *
+     * @return self
+     */
+    public function closeButtonSize(string $value): self
+    {
+        if (!in_array($value, self::SIZE_ALL, true)) {
+            $values = implode(' ', self::SIZE_ALL);
```
Comment (samdark) [inline src/ModalCard.php:182]:
```suggestion
            $values = implode('", "', self::SIZE_ALL);
```

File: src/ModalCard.php [hunk 12]
```diff
@@ -42,302 +46,495 @@ final class ModalCard extends Widget
...
 
         return $new;
     }
 
     /**
-     * Returns a new instance with the specified footer content.
+     * Returns a new instance with the specified toggle button color.
      *
-     * @param string $value The footer content in the modal window.
+     * @param string $value The toggle button color. Default setting empty without any color.
```
Comment (samdark) [inline src/ModalCard.php:444]:
```suggestion
     * @param string $value The toggle button color. By default there's no color set.
```

File: src/ModalCard.php [hunk 13]
```diff
@@ -42,302 +46,495 @@ final class ModalCard extends Widget
...
+     * Possible values are: ModalCard::COLOR_PRIMARY, ModalCard::COLOR_INFO, ModalCard::COLOR_SUCCESS,
+     * ModalCard::COLOR_WARNING, ModalCard::COLOR_DANGER, ModalCard::COLOR_DARK
      *
      * @return self
      */
-    public function footer(string $value): self
+    public function toggleButtonColor(string $value): self
     {
+        if (!in_array($value, self::COLOR_ALL, true)) {
+            $values = implode(' ', self::COLOR_ALL);
```
Comment (samdark) [inline src/ModalCard.php:453]:
```suggestion
            $values = implode('", "', self::COLOR_ALL);
```

File: src/ModalCard.php [hunk 14]
```diff
@@ -42,302 +46,495 @@ final class ModalCard extends Widget
...
-     * @param string $value The title content in the modal window.
+     * @param string $value The toggle button size.
      *
      * @return self
      */
-    public function title(string $value): self
+    public function toggleButtonSize(string $value): self
     {
+        if (!in_array($value, self::SIZE_ALL, true)) {
+            $values = implode('', self::SIZE_ALL);
```
Comment (samdark) [inline src/ModalCard.php:502]:
```suggestion
            $values = implode('", "', self::SIZE_ALL);
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: docs/modalcard.md (hunk 2 above)
Comment (terabytesoftw) [reply docs/modalcard.md:106]: done.

File: docs/modalcard.md (hunk 3 above)
Comment (terabytesoftw) [reply docs/modalcard.md:109]: They are not the same, one is the content, the other is the CSS class.

File: docs/modalcard.md (hunk 4 above)
Comment (terabytesoftw) [reply docs/modalcard.md:112]: Could merge it into one, but it's not convenient, if you just want to change the CSS class.

File: docs/modalcard.md (hunk 5 above)
Comment (terabytesoftw) [reply docs/modalcard.md:114]: It's the CSS Class of the `ModalCard`, as I call it CardCssClass ?

File: docs/modalcard.md (hunk 6 above)
Comment (terabytesoftw) [reply docs/modalcard.md:119]: done.

File: docs/modalcard.md (hunk 7 above)
Comment (terabytesoftw) [reply docs/modalcard.md:122]: done.

File: docs/modalcard.md (hunk 8 above)
Comment (terabytesoftw) [reply docs/modalcard.md:123]: done.

File: docs/modalcard.md (hunk 9 above)
Comment (terabytesoftw) [reply docs/modalcard.md:124]: done.

File: src/ModalCard.php (hunk 10 above)
Comment (terabytesoftw) [reply src/ModalCard.php:174]: done.

File: src/ModalCard.php (hunk 11 above)
Comment (terabytesoftw) [reply src/ModalCard.php:182]: done.

File: src/ModalCard.php (hunk 12 above)
Comment (terabytesoftw) [reply src/ModalCard.php:444]: done.

File: src/ModalCard.php (hunk 13 above)
Comment (terabytesoftw) [reply src/ModalCard.php:453]: done.

File: src/ModalCard.php (hunk 14 above)
Comment (terabytesoftw) [reply src/ModalCard.php]: done.

File: docs/modalcard.md (hunk 4 above)
Comment (samdark) [reply docs/modalcard.md:112]: No, I don't mean merging it to the same method. I mean naming i.e.:

```php
headerClass(string $value)
headerAttributes(array $value)
```

File: docs/modalcard.md (hunk 3 above)
Comment (samdark) [reply docs/modalcard.md:109]: Yes, got that. The question was if that is the same element referenced using two different names "foot" and "footer" or these are two different elements?

File: docs/modalcard.md (hunk 5 above)
Comment (samdark) [reply docs/modalcard.md:114]: Yes. That's better name 👍

Comment (terabytesoftw) [reply docs/modalcard.md]: done.

File: docs/modalcard.md (hunk 4 above)
Comment (terabytesoftw) [reply docs/modalcard.md]: done.

File: docs/modalcard.md (hunk 3 above)
Comment (terabytesoftw) [reply docs/modalcard.md]: done.

**Comment by samdark** [comment]
👍

**Pull request merged**